server/protocols/jmap/src/main/java/org/apache/james/jmap/exceptions/InvalidOriginMessageForMDNNotificationException.java

```diff
@@ -0,0 +1,44 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.exceptions;
+
+public class InvalidOriginMessageForMDNNotificationException extends Exception {
```

Comment:
s/MDNNotification/MDN/ ?

Reply:
Missing(Sender|Origin) i/of Invalid ?

Reply:
@fvn-linagora  the idea here is to be flexible, and assume the exception class name should say that the origin message is invalid

This exception is designed to be more flexible than just "X is missing".

Reply:
I'd say *YAGNI* ;) Better be explicit with guard clauses, in general. Only found usages through factory method `missingField`